From: Michal Kubecek <mkubecek@suse.cz>
Date: Mon, 27 May 2019 07:24:07 +0200
Subject: kvm: memunmap() also needs HAS_IOMEM
Patch-mainline: Submitted - 2019-05-27
References: none

Commit d30b214d1d0a ("kvm: fix compilation on s390") addresses link error
(undefined reference to memremap) without HAS_IOMEM but memunmap() is also
only available with HAS_IOMEM enabled so that we need similar fix in
kvm_vcpu_unmap().

Fixes: e45adf665a53 ("KVM: Introduce a new guest mapping API")
Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 virt/kvm/kvm_main.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -1795,8 +1795,10 @@ void kvm_vcpu_unmap(struct kvm_vcpu *vcpu, struct kvm_host_map *map,
 
 	if (map->page)
 		kunmap(map->page);
+#ifdef CONFIG_HAS_IOMEM
 	else
 		memunmap(map->hva);
+#endif
 
 	if (dirty) {
 		kvm_vcpu_mark_page_dirty(vcpu, map->gfn);
